src/store.ts

```diff
@@ -15,31 +15,94 @@ import userReducer, { initialUserState } from './auth/ducks/reducers';
+import {
+  TeamLeaderboardWeeklyReducerState,
+  TeamLeaderboardMonthlyReducerState,
+  TeamLeaderboardYearlyReducerState,
+  TeamLeaderboardAllTimeReducerState,
+} from './containers/team-leaderboard/ducks/types';
+import teamLeaderboardWeeklyReducer, {
+  initialTeamLeaderboardWeeklyState,
+} from './containers/team-leaderboard/ducks/weeklyReducer';
+import teamLeaderboardMonthlyReducer, {
+  initialTeamLeaderboardMonthlyState,
+} from './containers/team-leaderboard/ducks/monthlyReducer';
+import teamLeaderboardYearlyReducer, {
+  initialTeamLeaderboardYearlyState,
+} from './containers/team-leaderboard/ducks/yearlyReducer';
+import teamLeaderboardAllTimeReducer, {
+  initialTeamLeaderboardAllTimeState,
+} from './containers/team-leaderboard/ducks/allTimeReducer';
 
 export interface C4CState {
   authenticationState: UserAuthenticationReducerState;
+  userLeaderboardWeeklyState: UserLeaderboardWeeklyReducerState;
+  userLeaderboardMonthlyState: UserLeaderboardMonthlyReducerState;
+  userLeaderboardYearlyState: UserLeaderboardYearlyReducerState;
+  userLeaderboardAllTimeState: UserLeaderboardAllTimeReducerState;
+  teamLeaderboardWeeklyState: TeamLeaderboardWeeklyReducerState;
+  teamLeaderboardMonthlyState: TeamLeaderboardMonthlyReducerState;
+  teamLeaderboardYearlyState: TeamLeaderboardYearlyReducerState;
+  teamLeaderboardAllTimeState: TeamLeaderboardAllTimeReducerState;
 }
 
 export interface Action<T, P> {
   readonly type: T;
   readonly payload: P;
 }
 
-export type C4CAction = UserAuthenticationActions;
+export type C4CAction = UserAuthenticationActions & LeaderboardItemActions;
 
-export type ThunkExtraArgs = UserAuthenticationExtraArgs;
+export type ThunkExtraArgs = UserAuthenticationExtraArgs & ApiExtraArgs;
 
 const reducers = combineReducers<C4CState, C4CAction>({
   authenticationState: userReducer,
+  userLeaderboardWeeklyState: userLeaderboardWeeklyReducer,
+  userLeaderboardMonthlyState: userLeaderboardMonthlyReducer,
+  userLeaderboardYearlyState: userLeaderboardYearlyReducer,
+  userLeaderboardAllTimeState: userLeaderboardAllTimeReducer,
+  teamLeaderboardWeeklyState: teamLeaderboardWeeklyReducer,
+  teamLeaderboardMonthlyState: teamLeaderboardMonthlyReducer,
+  teamLeaderboardYearlyState: teamLeaderboardYearlyReducer,
+  teamLeaderboardAllTimeState: teamLeaderboardAllTimeReducer,
 });
 
 export const initialStoreState: C4CState = {
   authenticationState: initialUserState,
+  userLeaderboardWeeklyState: initialUserLeaderboardWeeklyState,
+  userLeaderboardMonthlyState: initialUserLeaderboardMonthlyState,
+  userLeaderboardYearlyState: initialUserLeaderboardYearlyState,
+  userLeaderboardAllTimeState: initialUserLeaderboardAllTimeState,
+  teamLeaderboardWeeklyState: initialTeamLeaderboardWeeklyState,
+  teamLeaderboardMonthlyState: initialTeamLeaderboardMonthlyState,
+  teamLeaderboardYearlyState: initialTeamLeaderboardYearlyState,
+  teamLeaderboardAllTimeState: initialTeamLeaderboardAllTimeState,
```

Comment:
Could you explain the reasoning for having a separate piece of state for each time period? In looking over this I see how it would be a bit confusing, as no reducers have more than one piece of state to them. 

I think there are two options here: 
- have a `userLeaderboardState` containing four sub-pieces of state, but all in one reducer. Store each slice of data within that reducer
- have a single `userLeaderboardState` that just stores one slice of data at a time. When a user changes the time period, we simply make a new request to the backend, and update the ONE piece of state to contain data for that time period only. 


Personally, I think option 2 is the cleanest solution. We don't have to worry about the data taking ~1 second to reload each time IMO. If implemented properly, we can just make the leaderboard capable of displaying _any_ range of data from today until X days in the past, though the UI doesn't have to reveal that.